sktime/base/tests/test_base.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for BaseObject universal base class.
+
+tests in this module:
+
+    test_get_class_tags - tests get_class_tags inheritance logic
+    test_get_class_tag  - tests get_class_tag logic, incl default value
+    test_get_tags       - tests get_tags inheritance logic
+    test_get_tag        - tests get_tag logic, incl default value
+    test_set_tags       - tests set_tags logic and related get_tags inheritance
+
+copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+
+__author__ = ["fkiraly"]
+
+from copy import deepcopy
+
+from sktime.base import BaseObject
+from sktime.utils._testing.deep_equals import deep_equals
+
+
+# Fixture class for testing tag system
+class FixtureClassParent(BaseObject):
+
+    _tags = {"A": "1", "B": 2, "C": [], 3: "D"}
+
+
+# Fixture class for testing tag system, child overrides tags
+class FixtureClassChild(FixtureClassParent):
+
+    _tags = {"A": 42, 3: "E"}
+
+
+FIXTURE_CLASSCHILD = FixtureClassChild
+
+FIXTURE_CLASSCHILD_TAGS = {"A": 42, "B": 2, "C": [], 3: "E"}
```

Comment:
Quick question, why do we test for lists and integers in tag values?

Reply:
I thought bool, int, and string are fine. I replaced the list.